File: src/main/java/no/fintlabs/kafka/topic/configuration/EventTopicConfigurationMappingService.java
package no.fintlabs.kafka.topic.configuration;

import org.springframework.stereotype.Service;

@Service
public class EventTopicConfigurationMappingService {

    public TopicConfiguration toTopicConfiguration(EventTopicConfiguration eventTopicConfiguration) {
        TopicConfiguration.TopicConfigurationBuilder topicConfigurationBuilder = TopicConfiguration.builder();

        topicConfigurationBuilder.partitions(eventTopicConfiguration.getPartitions());

        TopicDeleteCleanupPolicyConfiguration.TopicDeleteCleanupPolicyConfigurationBuilder
                deleteCleanupPolicyTopicConfigurationBuilder = TopicDeleteCleanupPolicyConfiguration.builder();

        deleteCleanupPolicyTopicConfigurationBuilder.retentionTime(eventTopicConfiguration.getRetentionTime());

        topicConfigurationBuilder.deleteCleanupPolicy(deleteCleanupPolicyTopicConfigurationBuilder.build());


        TopicSegmentConfiguration.TopicSegmentConfigurationBuilder segmentTopicConfigurationBuilder
                = TopicSegmentConfiguration.builder();

        segmentTopicConfigurationBuilder.openSegmentDuration(
                eventTopicConfiguration.getCleanupFrequency().getSegmentDuration()
        );

        topicConfigurationBuilder.segmentConfiguration(
                segmentTopicConfigurationBuilder.build()
        );

        return topicConfigurationBuilder.build();
    }
}

File: src/main/kotlin/no/novari/kafka/topic/configuration/EventTopicConfigurationMappingService.kt
package no.novari.kafka.topic.configuration

import org.springframework.stereotype.Service

@Service
class EventTopicConfigurationMappingService {
    fun toTopicConfiguration(eventTopicConfiguration: EventTopicConfiguration): TopicConfiguration {
        val topicConfigurationBuilder = TopicConfiguration.builder()
        topicConfigurationBuilder.partitions(eventTopicConfiguration.partitions)

        topicConfigurationBuilder.deleteCleanupPolicy(
            TopicDeleteCleanupPolicyConfiguration
                .builder()
                .retentionTime(eventTopicConfiguration.retentionTime)
                .build(),
        )

        topicConfigurationBuilder.segmentConfiguration(
            TopicSegmentConfiguration
                .builder()
                .openSegmentDuration(eventTopicConfiguration.cleanupFrequency.segmentDuration)
                .build(),
        )

        return topicConfigurationBuilder.build()
    }
}
